cmd/compute-domain-kubelet-plugin/main.go

```diff
@@ -202,9 +203,8 @@ func StartPlugin(ctx context.Context, config *Config) error {
 		return fmt.Errorf("path for cdi file generation is not a directory: '%v'", config.flags.cdiRoot)
 	}
 
-	// Setup signal handling for graceful shutdown
-	sigs := make(chan os.Signal, 1)
-	signal.Notify(sigs, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+	ctx, stop := signal.NotifyContext(ctx, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
```

Comment:
Just for the record, as I am re-reading about this.

`stop` is a function, and elsewhere in this code base I think we typically call it `cancel()`.

From stdlib docs:

> The stop function unregisters the signal behavior, which, like [signal.Reset](https://pkg.go.dev/os/signal#Reset), may restore the default behavior for a given signal. For example, the default behavior of a Go program receiving [os.Interrupt](https://pkg.go.dev/os#Interrupt) is to exit. Calling NotifyContext(parent, os.Interrupt) will change the behavior to cancel the returned context. Future interrupts received will not trigger the default (exit) behavior until the returned stop function is called.

Reply:
I am not sure whats the suggestion here? is it naming consistency? Are you both suggesting to replace stop with cancel?

Reply:
Lets change this to cancel for consistency with the rest of this component